firedrake/mg/embedded.py

```diff
@@ -26,6 +26,7 @@ class Cache(object):
         :arg element: The element to use for the caching."""
         def __init__(self, element):
             self.embedding_element = get_embedding_dg_element(element)
+            self._dat_versions = {}
```

Comment:
I think best practice is to [document these things as attributes of the class](https://numpydoc.readthedocs.io/en/latest/format.html#class-docstring), but I don't think that enough people are bothered about this for it to be necessary here. Perhaps a comment?